web/tests/pm_list_data.test.js

```diff
@@ -159,6 +159,15 @@ test("get_conversations", ({override}) => {
     set_pm_with_filter("iago@zulip.com");
     pm_data = pm_list_data.get_conversations();
     assert.deepEqual(pm_data, expected_data);
+
+    // Iago should also be part of the expected data, since we have set the DM filter to Iago.
+    pm_data = pm_list_data.get_conversations("Ali");
+    assert.deepEqual(
+        pm_data,
+        expected_data.filter(
+            (item) => item.recipients === "Alice, Bob" || item.recipients === "Iago",
+        ),
```

Comment:
I could have filtered out by `Me, myself` recipient, but I feel this is more correct if there are any other additions to the expected data in the future.

Reply:
Yeah, this is better. Ideally we'd extend the test data a little bit to have the logic verify how we handle cases that you might wonder about, such as:
- diacritics/accent marks
- email addresses that match the keyword, when the name doesn't.
- matches within a word; `salivar` doesn't match `ali` with this algorithm, and documenting that with a test is helpful.

Not trying to change what we actually do here in this comment, just explaining it'd be helpful to have the tests verify the cases one might wonder about, so that any logic changes that impact those decisions are explicitly visible to developers and reviewers through the test failing/needing changes.

Reply:
Since we're using `filter_people_by_search_terms` now for filtering, and that function is properly tested in node tests, I've just stubbed out that function for this PR's tests.